generate/generate.js

```diff
@@ -0,0 +1,253 @@
+const https = require("https");
+const fs = require("fs");
+
+// Get slug as argument
+const slug = process.argv[2];
+
+// Download the canonical data
+let dataUrl = `https://raw.githubusercontent.com/exercism/problem-specifications/main/exercises/${slug}/canonical-data.json`;
+https
+  .get(dataUrl, (res) => {
+    let body = "";
+    res.on("data", (chunk) => (body += chunk));
+    res.on("end", () => {
+      try {
+        let canonicalData = JSON.parse(body);
+        generateFiles(slug, canonicalData);
+      } catch (error) {
+        console.error(error.message);
+      }
+    });
+  })
+  .on("error", (error) => {
+    console.error(error.message);
+  });
+
+// Use the canonical data to generate the multiple files
+function generateFiles(slug, canonicalData) {
+  const exercise = kebabToPascal(slug);
+
+  let extractedFunctions = {};
+  extractFunctions(canonicalData, extractedFunctions);
+
+  const functionList = Object.entries(extractedFunctions)
+    .map(([key, _]) => key)
+    .join(", ");
+
+  const allFunctionsCode = Object.entries(extractedFunctions)
+    .map(([key, value]) => generateFunctionCode(key, value))
+    .join("\n\n");
+
+  const mainFile = `
+module ${exercise} exposing (${functionList})
+
+${allFunctionsCode}
+`;
+  console.log(mainFile);
+
+  const allTestCode = generateAllTestsCode(
+    exercise,
+    extractedFunctions,
+    canonicalData
+  );
+
+  const testFile = `
+module Tests exposing (tests)
+
+import ${exercise}
+import Expect
+import Test exposing (Test, describe, skip, test)
+
+${toElmComments(canonicalData.comments)}
+
+tests : Test
+tests = describe "${exercise}" [ ${allTestCode} ]
+`;
+
+  console.log(testFile);
+}
+
+// Generate the template code for one function
+function generateFunctionCode(name, { args, canError }) {
+  const returnType = canError ? "Result String Todo" : "Todo";
+  const typeAnnotation = [...Array(args.length).fill("Todo"), returnType].join(
```

Comment:
What does the `...` notation do? I know I should google it haha :D

Reply:
It's just syntax to concatenate an array with other values easily

```
let a = [ 1, 2, 3 ]
let b = [ ...a, 4, 5 ]
```